apps/dcbustimes/dc_bus_times.star

```diff
@@ -0,0 +1,257 @@
+                        Bus[1],
+                        Bus[2],
+                        Bus[3],
+                        Bus[4],
+                        Bus[5],
+                    ],
+                ),
+            ),
+        )
+    elif numPredictions == 7:
+        return render.Root(
+            delay = 500,
+            child = render.Marquee(
+                scroll_direction = "vertical",
+                height = 32,
+                align = "start",
+                offset_start = 2,
+                offset_end = 32,
+                child = render.Column(
+                    children = [
+                        Bus[0],
+                        Bus[1],
+                        Bus[2],
+                        Bus[3],
+                        Bus[4],
+                        Bus[5],
+                        Bus[6],
+                    ],
+                ),
+            ),
+        )
+    else:
+        return render.Root(
+            delay = 500,
+            child = render.Marquee(
+                scroll_direction = "vertical",
+                height = 32,
+                align = "start",
+                offset_start = 2,
+                offset_end = 32,
+                child = render.Column(
+                    children = [
+                        Bus[0],
+                        Bus[1],
+                        Bus[2],
+                        Bus[3],
+                        Bus[4],
+                        Bus[5],
+                        Bus[6],
+                        Bus[7],
+                    ],
+                ),
+            ),
+        )
+
+def GetTimes1(stopID, apiKey):
+    cached = cache.get("arrTimes1")
+    if cached:
+        return json.decode(cached)
+
+    rep = http.get(NEXTBUS_URL, params = {"StopID": stopID}, headers = {"api_key": apiKey})
+    if rep.status_code != 200:
+        fail("NextBus request failed with status ", rep.status_code)
+
+    cache.set("arrTimes1", rep.body(), ttl_seconds = 20)
+    return rep.json()
+
+def GetTimes2(stopID, apiKey):
+    cached = cache.get("arrTimes2")
+    if cached:
+        return json.decode(cached)
+
+    rep = http.get(NEXTBUS_URL, params = {"StopID": stopID}, headers = {"api_key": apiKey})
+    if rep.status_code != 200:
+        fail("NextBus request failed with status ", rep.status_code)
+
+    cache.set("arrTimes2", rep.body(), ttl_seconds = 20)
+    return rep.json()
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "StopID_1",
+                name = "Stop ID #1",
+                desc = "Bus Stop ID Number (7 digit number located on Bus Stop sign e.g. 1001155)",
+                icon = "busSimple",
+            ),
+            schema.Text(
+                id = "StopID_2",
+                name = "Stop ID #2",
+                desc = "Bus Stop ID Number (optional)",
+                icon = "busSimple",
+            ),
+            schema.Text(
```

Comment:
The `desc` field is currently not visible to end users in the phone app, so if the WMATA Key is optional it could be good to mention that in the field name (which is visible to users). Alternatively, since there's an encrypted key already available, perhaps this config field isn't needed at all?